extensions/http/jetty/src/main/java/org/eclipse/dataspaceconnector/extension/jetty/JettyConfiguration.java

```diff
@@ -14,20 +14,91 @@
 
 package org.eclipse.dataspaceconnector.extension.jetty;
 
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.system.Config;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 public class JettyConfiguration {
 
-    private final int httpPort;
+    public static final String WEB_HTTP_PREFIX = "web.http";
+    public static final String DEFAULT_PATH = "/api";
+    public static final String DEFAULT_CONTEXT_NAME = "default";
+    public static final int DEFAULT_PORT = 8181;
+    @EdcSetting
+    private static final String HTTP_PORT = "web.http.port";
     private final String keystorePassword;
     private final String keymanagerPassword;
+    private final Set<PortMapping> portMappings;
 
-    public JettyConfiguration(int httpPort, String keystorePassword, String keymanagerPassword) {
-        this.httpPort = httpPort;
+    public JettyConfiguration(String keystorePassword, String keymanagerPassword) {
         this.keystorePassword = keystorePassword;
         this.keymanagerPassword = keymanagerPassword;
+        portMappings = new HashSet<>();
     }
 
-    public int getHttpPort() {
-        return httpPort;
+    public static JettyConfiguration createFromConfig(String keystorePassword, String keymanagerPassword, Config config) {
+        var jettyConfig = new JettyConfiguration(keystorePassword, keymanagerPassword);
+
+        var subConfig = config.getConfig(WEB_HTTP_PREFIX);
+
+        Map<String, Map<String, String>> tempMappings = new HashMap<>();
```

Comment:
This logic (without supporting multipathconfig, that in my opinion are not really useful as I commented in the test class) could be handled by the `Config` object adding a simple `isLeaf` method on it, that tells you if there's only one entry and its path equals root path and using this code snipped instead:

```
Stream.concat(Stream.of(subConfig), subConfig.partition())
                .filter(it -> !it.isLeaf())
                .map(c -> {
                    String currentNode = c.currentNode();
                    if (currentNode.equals("default")) throw new IllegalArgumentException("A 'default' port mapping cannot be defined");

                    var name = currentNode.equals("http") ? "default" : currentNode;
                    return new PortMapping(name, c.getInteger("port", DEFAULT_PORT), c.getString("path", DEFAULT_PATH));
                })
                .forEach(jettyConfig::portMapping);
```

Reply:
I quickly threw this into `ConfigImpl` assuming that's what you meant:
```java
    @Override
    public boolean isLeaf() {
        return entries.size() == 1 && entries.keySet().stream().allMatch(rootPath::equals);
    }
```
The only thing which then won't work is something like `web.http.this.is.a.long.port`, where `this.is.a.long` would previously have been regarded as the context alias.
That's a minor drawback though, so I'm good with adopting your solution.

Reply:
yes, it does, is the 
```var name = currentNode.equals("http") ? "default" : currentNode;```
line that does the job

Reply:
@ndr-brt are you OK with merging this as-is and investigate an improvement down the road? This PR is required for a bunch of others, and MS2 is fast approaching :)

Reply:
ok